airtime_mvc/application/common/ProvisioningHelper.php

```diff
@@ -34,34 +35,63 @@ public function createAction()
         try {
 
             $this->parsePostParams();
-            
+
             //For security, the Airtime Pro provisioning system creates the database for the user.
-            $this->setNewDatabaseConnection();
+            if ($this->dbhost && !empty($this->dbhost)) {
+                $this->setNewDatabaseConnection();
 
-            //if ($this->checkDatabaseExists()) {
-            //    throw new Exception("ERROR: Airtime database already exists");
-            //}
+                //if ($this->checkDatabaseExists()) {
+                //    throw new Exception("ERROR: Airtime database already exists");
+                //}
 
-            if (!$this->checkDatabaseExists()) {
-                throw new Exception("ERROR: $this->dbname database does not exist.");
-            }
+                if (!$this->checkDatabaseExists()) {
+                    throw new Exception("ERROR: $this->dbname database does not exist.");
+                }
 
-            //We really want to do this check because all the Propel-generated SQL starts with "DROP TABLE IF EXISTS".
-            //If we don't check, then a second call to this API endpoint would wipe all the tables!
-            if ($this->checkTablesExist()) {
-                throw new Exception("ERROR: airtime tables already exists");
+                //We really want to do this check because all the Propel-generated SQL starts with "DROP TABLE IF EXISTS".
+                //If we don't check, then a second call to this API endpoint would wipe all the tables!
+                if ($this->checkTablesExist()) {
+                    throw new Exception("ERROR: airtime tables already exists");
+                }
+
+                $this->createDatabaseTables();
+                $this->initializeMusicDirsTable($this->instanceId);
             }
 
             //$this->createDatabase();
 
             //All we need to do is create the database tables.
 
-            $this->createDatabaseTables();
-            $this->initializeMusicDirsTable($this->instanceId);
+            $this->initializePrefs();
```

Comment:
There needs to be some security here to prevent this action from being called on a station that is already configured.... (If we have an airtime.conf, return?)